src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -3955,9 +3958,6 @@ class CL1
                 // (27,16): error CS0165: Use of unassigned local variable 'x3'
                 //         M2(ref x3);
                 Diagnostic(ErrorCode.ERR_UseDefViolation, "x3").WithArguments("x3").WithLocation(27, 16),
-                // (27,16): warning CS8601: Possible null reference assignment.
```

Comment:
>// (27,16): warning CS8601: Possible null reference assignment. [](start = 16, length = 63)

It feels like this warning should still be valid.

Reply:
`x3` is unassigned so its nullability state is `null`.

Reply:
> x3 is unassigned so its nullability state is null.

I am not sure if that should make any difference. Since we passed it by ref, null value can be written to it by the method and it looks like that was the reason for the warning. That part didn't change

---
In reply to: [175649604](https://github.com/dotnet/roslyn/pull/25411#discussion_r175649604) [](ancestors = 175649604)

Reply:
Also it would be good to add similar test for compound assignment and ```++```.

---
In reply to: [175651207](https://github.com/dotnet/roslyn/pull/25411#discussion_r175651207) [](ancestors = 175651207,175649604)

Reply:
Added the diagnostic back with a `// PROTOTYPE` comment, and added similar cases to `CompoundAssignment_04` and `IncrementOperator_01`.